File: chars/CharArithmeticDemo.java
import static java.lang.String.format;

final class CharArithmeticDemo {

    public static void run() {
        char ch = 'X';
        System.out.println(format("Initial value: %c", ch));

        ch++;
        System.out.println(format("After increment: %c", ch));  // Z

        ch = 90;
        System.out.println(format("After setting 90: %c", ch)); // Z
    }
}

File: chars/CharArithmeticDemo.cs
using static System.Console;

public class CharArithmeticDemo
{
    public static void Run()
    {
        var ch = 'X';
        WriteLine($"Initial value: {ch}");

        ch++;
        WriteLine($"After increment: {ch}");

        ch = (char) 90;
        WriteLine($"After setting 90: {ch}");
    }
}
